website/server/controllers/top-level/auth.js

```diff
@@ -2,45 +2,94 @@ import locals from '../../middlewares/locals';
 import { decrypt } from '../../libs/encryption';
 import moment from 'moment';
 import { model as User } from '../../models/user';
+import * as passwordUtils from '../../libs/password';
 
 let api = {};
 
 // Internal authentication routes
 
-// Set a new password after having requested a password reset
+// TODO
+// - convert to bcrypt
+// - tests
+
+// returns the user if a valid password reset code is supplied, otherwise false
+async function validatePasswordResetCodeAndFindUser (code) {
```

Comment:
You should move these functions out to a library (maybe the password utils) so they can be tested.